src/pretix/plugins/sendmail/models.py

```diff
@@ -97,14 +98,33 @@ def recompute(self):
             )
             if not datetime_exists(self.computed_datetime):
                 self.computed_datetime = make_aware(
-                    datetime.combine(d, time(hour=st.hour, minute=st.minute, second=st.second, microsecond=0)) + timedelta(hours=1),
+                    datetime.combine(d,
+                                     time(hour=st.hour, minute=st.minute, second=st.second, microsecond=0)) + timedelta(
+                        hours=1),
                     self.event.timezone,
                 )
 
         if self.computed_datetime > timezone.now() and self.state == self.STATE_MISSED:
             self.state = self.STATE_SCHEDULED
         self.last_computed = timezone.now()
 
+    def filter_checkins(self, limit_products):
+        filter_positions = Q()
+        if self.subevent:
+            filter_positions |= Exists(OrderPosition.objects.filter(order=OuterRef('pk'), subevent=self.subevent))
+        if not self.rule.all_products:
+            filter_positions |= Exists(OrderPosition.objects.filter(order=OuterRef('pk'), item_id__in=limit_products))
+
+        checkin_q = Q()
+        if self.rule.checked_in_status == "no_checkin":
+            checkin_q = ~Exists(
+                Checkin.objects.filter(position__order_id=OuterRef('pk'),
+                                       position_id__in=OrderPosition.objects.filter(filter_positions)))
+        if self.rule.checked_in_status == "checked_in":
+            checkin_q = Exists(Checkin.all.filter(position__order_id=OuterRef('pk'),
+                                                  position_id__in=OrderPosition.objects.filter(filter_positions)))
+        return checkin_q
```

Comment:
This would need rewriting to filter directly on `orders` queryset, e.g. like so (also I think we do not need the filter on OrderPosition as that should already be filtered by line 138 ff and 145 ff?):

```
       if self.rule.checked_in_status == "no_checkin": 
           orders = orders.filter(
                Exists(Checkin.objects.filter(position__order_id=OuterRef('pk'))
            )
        elif self.rule.checked_in_status …
            …
        return orders
```